### ISSUE TITLE
Provide rpc server/client/dispatcher

### ISSUE THREAD

**Issue by harlowja** [celery/kombu#430]
To make kombu more useful for users wishing to use it as a rpc server/dispatcher/client it would be really neat to provide a foundational support for this kind of code in kombu itself (in a way similar to the `async` module in kombu which adds/aids in `asyncio` support in kombu, but perhaps named `rpc` instead). 

This would start to remove some of the openstacks code base (called oslo.messaging which provides these features, mainly ontop of kombu, the current code + other code this library provides is @ https://github.com/openstack/oslo.messaging/tree/master/oslo/messaging). It'd be nice to start to move some of that code upstream into kombu so that the wider python community gets to take advantage of these types of features (instead of just the oslo/openstack community).

I'd like to start this kind of discussion and potentially see what the kombu folks response would be to this proposal (starting with moving the `rpc` code in oslo.messaging to kombu, or a refactored version thereof - https://github.com/openstack/oslo.messaging/tree/master/oslo/messaging/rpc).

As this further progresses we can move more of the oslo.messaging code into kombu (notifications @ https://github.com/openstack/oslo.messaging/tree/master/oslo/messaging/notify might be yet another simple addition that would provide some meaningful value to kombu).

**TLDR**
- standard `rpc` kombu provided module
- helps create a common module, avoiding the need to re-implement the same thing in multiple projects (such as https://github.com/ceelian/callme or https://github.com/openstack/oslo.messaging ...)
- world peace...

Thoughts?

P.S. If there is a better place to discuss this (mailing list, launchpad...) let me know and I'll move this there :smile:. Thanks!
